packages/grid/src/Grid.tsx

```diff
@@ -549,6 +560,49 @@ class Grid extends PureComponent<GridProps, GridState> {
     if (this.validateSelection()) {
       this.checkSelectionChange(prevState);
     }
+
+    const {
+      rowHeight,
+      bottom,
+      visibleRowYs,
+      visibleRowHeights,
+      columnHeaderHeight,
+    } = this.metrics;
+    const arr: [number, number][] = [];
+    userRowHeights.forEach((x, number) => arr.push([x, number]));
+    if (updateTop) {
```

Comment:
Instead of putting this in `componentDidUpdate`, this logic should either be in the `moveSelection` method (with some sort of `viewportPosition` parameter added) or a new method. Then IrisGrid calls `grid.moveSelection`